src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnectionFactory.java

```diff
@@ -83,21 +97,42 @@ public ServiceBusJmsConnectionFactory(ConnectionStringBuilder connectionStringBu
      */
     public ServiceBusJmsConnectionFactory(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
         this.settings = settings;
+        this.sasKey = sasKey;
+        this.sasKeyName = sasKeyName;
         this.initialize(sasKeyName, sasKey, host, settings);
     }
     
-    private void initialize(String sasKeyName, String sasKey, String host, ServiceBusJmsConnectionFactorySettings settings) {
-        if (sasKeyName == null || sasKeyName == null || host == null) {
-            throw new IllegalArgumentException("SAS Key, SAS KeyName and the host cannot be null for a ServiceBus connection factory.");
+    /**
+     * Create a ServiceBusJmsConnectionFactory using a credential and host name.
+     * @param Credential. A token provider credential that will be use to acquire an aad token. For default credential ether pass null or DefaultAzure
+     * @param host The host name of the ServiceBus namespace. Example: your-namespace-name.servicebus.windows.net
+     * @param settings The options used for this ConnectionFactory. Null can be used as default.
+     * @throws Exception 
+     */
+    public ServiceBusJmsConnectionFactory(TokenCredential credential, String host, ServiceBusJmsConnectionFactorySettings settings){
+    	this.settings = settings;
+    	this.isAadAuthentication =true;
```

Comment:
>this.isAadAuthentication =true; [](http://example.com/codeflow?start=5&length=31)

Can we use the typeof Object to infer this instead?

If Object Type is AadAuthentication or the one for DefaultAzureCredential, then intialize with AAD credentials instead.